CHANGELOG.md

```diff
@@ -7,10 +7,16 @@ Please enumerate all user-facing changes using format `<githib issue/pr number>:
 ## 0.9.0
 
 
-
+* [#322](https://github.com/kroxylicious/kroxylicious-junit5-extension/pull/322): Allow test Kafka Clusters to use SASL SCRAM-SHA and OAUTH bearer
 * [#289](https://github.com/kroxylicious/kroxylicious-junit5-extension/pull/289): Bump org.testcontainers:testcontainers-bom from 1.19.6 to 1.19.7
 * [#283](https://github.com/kroxylicious/kroxylicious-junit5-extension/pull/283): Bump kafka.version from 3.6.1 to 3.7.0.
 
+### Changes, deprecations and removals
+
+* For SCRAM support in KRaft mode,  the Kafka Broker must be 3.5 or above and it must use storage format IBP_3_5_IV2
+  or higher.  If using CONTAINER mode, be aware that support for this combination was introduced in
```

Comment:
If the user uses an older kafka-native container image without this support and tries to create scram users, it'll fail to startup.  The failure is obvious.  There's no impact for users using SASL PLAIN.

Reply:
Am I right in saying we can't automate this to fail fast for invalid combinations as we can't tell what the broker configuration is.

Reply:
We could read the IBP version and fail if it doesn't support SCRAM.   In the containers case, we'd need to start the broker first then sniff using Admin client.  If the IBP version is too low, and the broker started, we could infer that the container didn't understand `SERVER_SCRAM_CREDENTIALS`.   It seemed like quite a bit of fiddly code.   I think anyone who wants to use SCRAM will be upgrading anyway.